### PR TITLE
Add animationClip support

### PR DESCRIPTION
**Pull request by sigvef**
It is gloriously underspecified at the moment, but it is at least a
starting point!

### CODE REVIEW

**Comment by sigvef** [comment]
To use, add something like this to a layer in res/layers.json:

```
"animationClips": {
        "glowAmount": [
            {"relativeFrame": 0, "value": 0, "type": "hold"},
            {"relativeFrame": 80, "value": 0, "type": "hold"},
            {"relativeFrame": 100, "value": 1, "type": "lerp"},
            {"relativeFrame": 150, "value": 0, "type": "smoothstep"},
            {"relativeFrame": 4000, "value": 0, "type": "hold"}
        ],  
        "bikeElevation": [
            {"relativeFrame": 0, "value": 0.5, "type": "hold"},
            {"relativeFrame": 100, "value": 0.5, "type": "hold"},
            {"relativeFrame": 120, "value": 1, "type": "smoothstep"},
            {"relativeFrame": 140, "value": 0, "type": "smoothstep"},
            {"relativeFrame": 160, "value": 1, "type": "smoothstep"},
            {"relativeFrame": 180, "value": 0, "type": "smoothstep"},
            {"relativeFrame": 200, "value": 0.5, "type": "smoothstep"},
            {"relativeFrame": 4000, "value": 0, "type": "hold"}
        ]   
    }
```

**Comment by sigvef** [comment]
This is what it looks like:

image

File: nin/dasBoot/LayerManager.js [hunk 1]
```diff
@@ -54,6 +54,28 @@ LayerManager.prototype.update = function(frame) {
...
+            }
+          }
+          if(from) {
+            li[key] = window[to.type](from.value, to.value,
+                                      (relativeFrame - from.relativeFrame) /
+                                      (to.relativeFrame - from.relativeFrame));
+          } else if(to) {
+            li[key] = to.value;
+          } else {
+            li[key] = from.value;
```
Comment (stianjensen) [inline nin/dasBoot/LayerManager.js]: As I read this, it seems that the only case when we reach this else is when both `from` and `to` are undefined. In that case, `from.value` won't work here.

**Comment by sigvef** [comment]
ping

**Comment by aleksanb** [comment]
How about Automation Clips?

**Comment by sigvef** [comment]
Continue discussion in #209.

**Pull request closed without merging**